### PR TITLE
concurrency: small tweaks to the atomics section

### PR DESCRIPTION
**Pull request by japaric**
r? @adamgreig

### CODE REVIEW

**Comment by adamgreig** [comment]
Oops, I should not have put them as `static mut` in the first place! Thanks.

I'm not _completely_ sure `Relaxed` is always sufficient on single-core platforms (what about Cortex-M7 with its data cache that needs `DMB` etc when DMA might read it? what about compiler re-ordering instructions around a lock atomic in the face of interrupts triggering?). I get that `Relaxed` is always fine for a counter (as in this example), but I thought that would be true even on multi-core systems.

Anyway I think the new text is fine too and really anyone needing to know about this probably needs to read more than this book, so.. lgtm.

File: src/concurrency/concurrency.md [hunk 1]
```diff
@@ -174,30 +174,30 @@ fn main() -> ! {
         if state && !last_state {
             last_state = state;
             // Use `fetch_add` to atomically add 1 to COUNTER
-            unsafe { COUNTER.fetch_add(1, atomic::Ordering::Relaxed) };
+            COUNTER.fetch_add(1, atomic::Ordering::Relaxed);
```
Comment (adamgreig) [inline src/concurrency/concurrency.md]: Should just be `Ordering::Relaxed`

File: src/concurrency/concurrency.md [hunk 2]
```diff
@@ -174,30 +174,30 @@ fn main() -> ! {
...
         }
     }
 }
 
 #[interrupt]
 fn timer() {
     // Use `store` to write 0 directly to COUNTER
-    unsafe { COUNTER.store(0, atomic::Ordering::Relaxed) }
+    COUNTER.store(0, atomic::Ordering::Relaxed)
```
Comment (adamgreig) [inline src/concurrency/concurrency.md]: `Ordering::Relaxed`

File: src/concurrency/concurrency.md (hunk 1 above)
Comment (japaric) [reply src/concurrency/concurrency.md]: fixed

**Pull request merged**